azure-functions/acbs-function/mappings/facility/helpers/get-exposure-period.js

```diff
@@ -54,40 +53,28 @@ const getExposurePeriod = (facility, dealType, fmr = null) => {
     }
 
     // If not already calculated
-    let coverStartDate;
-    const {
-      requestedCoverStartDate,
-      ukefGuaranteeInMonths,
-    } = facilitySnapshot;
-
-    if (requestedCoverStartDate) {
-      const startDate = moment(formatTimestamp(requestedCoverStartDate));
-
-      coverStartDate = moment([
-        formatYear(startDate.year()),
-        startDate.month(),
-        startDate.date(),
-      ]);
-    } else {
-      coverStartDate = moment([
+    const { requestedCoverStartDate, ukefGuaranteeInMonths } = facilitySnapshot;
+
+    const coverStartDate = requestedCoverStartDate
+      ? formatTimestamp(requestedCoverStartDate)
+      : getDateStringFromYearMonthDay(
         formatYear(facilitySnapshot['requestedCoverStartDate-year']),
-        facilitySnapshot['requestedCoverStartDate-month'] - 1,
+        facilitySnapshot['requestedCoverStartDate-month'],
         facilitySnapshot['requestedCoverStartDate-day'],
-      ]);
-    }
+      );
 
-    const coverEndDate = moment([
+    const coverEndDate = getDateStringFromYearMonthDay(
       formatYear(facilitySnapshot['coverEndDate-year']),
-      facilitySnapshot['coverEndDate-month'] - 1,
+      facilitySnapshot['coverEndDate-month'],
       facilitySnapshot['coverEndDate-day'],
-    ]);
+    );
 
-    if ((!coverStartDate.isValid() || !coverEndDate.isValid()) && ukefGuaranteeInMonths) {
+    if ((!isDate(coverStartDate) || !isDate(coverEndDate)) && ukefGuaranteeInMonths) {
       exposure = ukefGuaranteeInMonths;
     }
 
-    const durationMonths = coverEndDate.diff(coverStartDate, 'months') + 1;
-    const monthOffset = moment(coverStartDate).date() === moment(coverEndDate).date() ? -1 : 0;
+    const durationMonths = getMonthDifference(coverStartDate, coverEndDate) + 1;
+    const monthOffset = isSameDayOfMonth(coverStartDate, coverEndDate) ? -1 : 0;
 
     // Return `exposure` when `coverStartDate` and `coverEndDate` are null
     if (!durationMonths && !monthOffset) {
```

Comment:
I dont know why this is calculated this way or the if statement is needed - ideally would like to replace with
```suggestion
    const durationMonths = getMonthDifference(coverStartDate, coverEndDate);
    const monthOffset = isSameDayOfMonth(coverStartDate, coverEndDate) ? 0 : 1;
```

Reply:
Would infact turn this into a single helper `getInclusiveMonthsDifference` and get rid of other two

Reply:
Yeah to me this sounds ideal to be honest, I don't mind if you do this now or in the next PR but +1 to it

Reply:
Documenting for my own sanity:

 - passing valid strings `yyyy-MM-dd` into getMonthDifference will return a number
 - passing `Invalid date` into getMonthDifference will return NaN
 - passing valid strings `yyyy-MM-dd` into isSameDayOfMonth will only return false if same day of month
 - passing `Invalid date` into isSameDayOfMonth will return fal
hence each is only falsy if:
 - same day of month & month difference = -1 (I'll assume month difference is positive)
 - either date is `Invalid date`
so this can be replaced with
```
if (coverStartDate === 'Invalid date' || coverEndDate === 'Invalid date') {
  return String(exposure);
}
```
but then the previous if statement is triggered